src/JsonProducer.php

```diff
@@ -48,6 +50,16 @@ public function publish(
     ) {
         $attributes = array_merge_recursive($this->defaultAttributes, $attributes);
 
-        $this->exchange->publish(json_encode($message), $routingKey, $flags, $attributes);
+        try {
+            $message = json_encode($message);
+        } catch (\Throwable $e) {
```

Comment:
To my knowledge `json_encode` will never throw an exception ([SO](https://stackoverflow.com/questions/12265830/can-json-encode-trigger-a-catch-block), [PHP Doc Comment](http://be2.php.net/manual/de/function.json-encode.php#117615). )